allensdk/test/brain_observatory/test_demixer.py

```diff
@@ -62,17 +62,25 @@ def test_demix_raises_warning_for_singular_matrix(
 
 
 @pytest.mark.parametrize(
-    "raw_traces,stack,masks,expected",
+    "raw_traces,stack,masks,block_size,expected",
     [
         (
             np.array([[2.0, 0.0], [2.0, 2.0]]),
             np.array([[[2., 2.], [2., 1.]], [[2., 2.], [2., 1.]]]),
             np.array([[[1, 0], [0, 0]], [[1, 1], [0, 0]]]),
+            1000,
+            (np.array([[0, 0], [2, 0]]), [False, True])
+        ),
+        (
+            np.array([[2.0, 0.0], [2.0, 2.0]]),
+            np.array([[[2., 2.], [2., 1.]], [[2., 2.], [2., 1.]]]),
+            np.array([[[1, 0], [0, 0]], [[1, 1], [0, 0]]]),
+            -1,  # stack processed in one block
             (np.array([[0, 0], [2, 0]]), [False, True])
         ),
     ],
 )
-def test_demix_time_dep_masks(raw_traces, stack, masks, expected):
-    result = dmx.demix_time_dep_masks(raw_traces, stack, masks)
+def test_demix_time_dep_masks(raw_traces, stack, masks, block_size, expected):
```

Comment:
Add a test for block_size < 1 and not == -1, to test case where exception is raised.
You can break into another testing function and use
```python

 with pytest.raises(ValueError, match=<expected_message>):
    result = dmx.demix_time_dep_masks(raw_traces, stack, masks, block_size, expected)

```
I want to see this so that we can capture that exceptions are being raised when we expect them to be raised.

Reply:
Very good, will implement this.